tests/rados/test_reads_balancer.py

```diff
@@ -0,0 +1,273 @@
+    regex = r"\s*(\d.\d)-rhel-\d"
+    build = (re.search(regex, config.get("build", config.get("rhbuild")))).groups()[0]
+    if not float(build) >= 7.0:
+        log.info(
+            "Test running on version less than 7.0, skipping verifying Reads Balancer functionality"
+        )
+        return 0
+
+    log.info("Starting the test to verify reads balancer functionality on RHCS cluster")
+    log.debug("Creating multiple pools for testing")
+    pools = config["create_pools"]
+    for each_pool in pools:
+        cr_pool = each_pool["create_pool"]
+        if cr_pool.get("pool_type", "replicated") == "erasure":
+            method_should_succeed(
+                rados_obj.create_erasure_pool, name=cr_pool["pool_name"], **cr_pool
+            )
+        else:
+            method_should_succeed(rados_obj.create_pool, **cr_pool)
+        method_should_succeed(rados_obj.bench_write, **cr_pool)
+    log.info("Completed creating pools and writing test data into the pools")
+
+    existing_pools = rados_obj.list_pools_on_cluster()
+
+    log.debug(
+        "Checking the scores on each pool on the cluster, if the pool is replicated pool"
+    )
+    for pool_name in existing_pools:
+        pool_details = rados_obj.get_pool_details(pool=pool_name)
+        log.debug(f"Selected pool name : {pool_name}")
+        if not pool_details["erasure_code_profile"]:
+            log.debug(
+                f"Selected pool is a replicated pool. Checking read balancer scores.\n"
+                f" Fetched from pool {pool_details['read_balance']}"
+            )
+            read_scores = pool_details["read_balance"]
+            if read_scores["score_acting"] > pool_details["size"]:
+                log.error(
+                    f"The balance score on the pool is more than the size of the pool."
+                    f"Balance score : {read_scores['score_acting']}, Size of poool : {pool_details['size']}"
+                )
+                log.error(
+                    "This failure will be ignored because of the bug "
+                    ": https://bugzilla.redhat.com/show_bug.cgi?id=2237352 and the failure will be added post fix"
+                )
+
+                # todo: uncomment the below exception once the bug is fixed
+                # raise Exception("Scores out of bound failure")
+
+            log.debug("Checking if the scores are calculated properly")
+            # Balance_score = raw_score / Optimal_score
+            expected_score = (
+                read_scores["raw_score_acting"] / read_scores["optimal_score"]
+            )
+            if round(read_scores["score_acting"], 2) != round(expected_score, 2):
+                log.error(
+                    f"The scores for the pool is not calculated properly. acting score {read_scores['score_acting']}"
+                    f"Expected : {(read_scores['raw_score_acting'] / read_scores['optimal_score'])}"
+                )
+                raise Exception("Scores out of bound failure")
```

Comment:
I'm unable to find a `try, except, finally` block to catch this exception. Please share the line# if I'm wrong

Reply:
I did not have one in the test. added.